src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -547,6 +547,23 @@ void BossAI::UpdateAI(uint32 diff)
     DoMeleeAttackIfReady();
 }
 
+void BossAI::_DespawnAtEvade()
+{
+    uint32 corpseDelay = me->GetCorpseDelay();
+    uint32 respawnDelay = me->GetRespawnDelay();
+
+    me->SetCorpseDelay(1);
```

Comment:
As per IRC discussion, this is to prevent a core crash from leaving the DB with a next-week respawn time (`->SetRespawnTime` does not save to DB).